pandas/tests/series/methods/test_convert_dtypes.py

```diff
@@ -206,7 +206,20 @@ def test_convert_dtypes(
         # Test that it is a copy
         copy = series.copy(deep=True)
 
-        result[result.notna()] = np.nan
+        if result.notna().sum() > 0 and result.dtype in [
```

Comment:
doesnt have to be for this PR, but long-term should we also warn/raise in the no-op case?

Reply:
as in, warn if
```python
ser[empty_mask] = incompatible_value
```
?

I wouldn't think that would be necessary, but we can discuss it